core/src/main/java/com/linecorp/armeria/common/util/BlockingTaskExecutorBuilder.java

```diff
@@ -103,6 +104,14 @@ public BlockingTaskExecutorBuilder priority(int priority) {
         return this;
     }
 
+    /**
+     * Configure if idle threads should be terminated after timeout.
+     */
+    public BlockingTaskExecutorBuilder allowThreadTimeOut(boolean value) {
+        allowThreadTimeOut = value;
+        return this;
+    }
+
```

Comment:
How about enabling this automatically if `keepAliveTime` is 0, instead of adding this method?

Reply:
Umm, I thought set keepAliveTime as 0 will make exception.
`java.lang.IllegalArgumentException: Core threads must have nonzero keep alive times`.

Reply:
> Umm, I thought set keepAliveTime as 0 will make exception.

Maybe we can set the default value for `keepAliveTime` to 0 and conditionally enable `allowThreadTimeout` flag if it has been overridden?

We should give default timeout for `CommonPools#BLOCKING_TASK_EXECUTOR` if we do so.

Reply:
Oh, sorry
The condition for this exception is `time == 0 && allowsCoreThreadTimeOut()`.

Reply:
@ks-yim Can we set `allowCoreThreadTimeOut` true when `keepAliveTime` is zero?
I've tried, but it doesn't work.😭 
```
    public void allowCoreThreadTimeOut(boolean value) {
        if (value && keepAliveTime <= 0)
            throw new IllegalArgumentException("Core threads must have nonzero keep alive times");
        if (value != allowCoreThreadTimeOut) {
            allowCoreThreadTimeOut = value;
            if (value)
                interruptIdleWorkers();
        }
    }
```

Reply:
I remove that method and set `allowCoreThreadTimeOut = true` when `keepAliveTime > 0`.
Please check again. 😄

Reply:
Yeah, in `ScheduledThreadPoolExecutor`, `keepAliveTime` does nothing when `allowCoreThreadTimeout` is `false`, so you don't really need to set it.